framework/src/org/checkerframework/framework/type/AnnotationClassLoader.java

```diff
@@ -483,6 +482,9 @@ private final void printPaths() {
      * @return the set of loaded annotation classes
      */
     public final Set<Class<? extends Annotation>> getLoadedAnnotationClasses() {
+        if (loadedAnnotations == null) {
```

Comment:
I would personally abstract this out into a getLoadedAnnotations method, because otherwise it could be easy to forget to do.  Then comment on line 96 that one should not use the field directly.